tdrs-backend/tdpservice/email/email.py

```diff
@@ -0,0 +1,88 @@
+"""Send emails."""
+
+from celery import shared_task
+from django.core.exceptions import ValidationError
+from django.core.validators import validate_email
+from django.core.mail import EmailMultiAlternatives
+from django.conf import settings
+from django.template.loader import get_template
+from django.template import TemplateDoesNotExist
+from tdpservice.email.email_enums import EmailType
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+@shared_task
+def construct_email(email_type, context):
+    """Get email template."""
+    template_path = email_type.value + ".html"
+    try:
+        template = get_template(template_path)
+    except TemplateDoesNotExist:
+        logger.error(f'Template {template_path} does not exist')
+        return
+
+    return template.render(context)
+
+@shared_task
+def mail(email_type: EmailType, recipient_email: str, email_context: dict = None) -> None:
+    """Send email to user."""
+    subject = email_context['subject']
+    html_message = construct_email(email_type, email_context)
+
+    if email_context['text_message']:
+        text_message = email_context['text_message']
+    else:
+        text_message = 'An email was sent with HTML content. Please view in an HTML capable email client.'
+
+    send_email(subject, text_message, html_message, [recipient_email])
+
+@shared_task
+def send_email(subject: str, message: str, html_message: str, recipient_list: list) -> bool:
+    """Send an email to a list of recipients."""
+    valid_emails = validate_emails(recipient_list)
+
+    # use EmailMultiAlternatives using the html_message and message
+
+    msg = EmailMultiAlternatives(subject, message, settings.DEFAULT_FROM_EMAIL, valid_emails)
+    msg.attach_alternative(html_message, "text/html")
+    response = msg.send()
+
+    if response == 0:
+        logger.error('Email failed to send')
+        return False
```

Comment:
This check should be here but maybe I should raise an error.

Reply:
Does returning `False` mark the job as failed in celery? I'd opt for throwing an error for consistency with the rest of the code but 🤷